apps/nowcasting-app/components/charts/delta-view/delta-view-chart.tsx

```diff
@@ -0,0 +1,755 @@
+  negative?: boolean;
+}> = ({ gspDeltas, setClickedGspId, negative = false }) => {
+  const [selectedBuckets] = useGlobalState("selectedBuckets");
+  const [selectedDeltaGsp, setSelectedDeltaGsp] = useGlobalState("selectedDeltaGsp");
+  const [clickedGspId] = useGlobalState("clickedGspId");
+  if (!gspDeltas.size) return null;
+
+  const sortFunc = (a: GspDeltaValue, b: GspDeltaValue) => {
+    if (negative) {
+      return a.delta - b.delta;
+    } else {
+      return b.delta - a.delta;
+    }
+  };
+
+  const deltaArray = Array.from(gspDeltas.values());
+
+  return (
+    <>
+      <div className={`flex flex-col flex-1 ${!negative ? "pl-3" : "pr-3 "}`}>
+        {deltaArray.sort(sortFunc).map((gspDelta) => {
+          let bucketColor = "";
+          let dataKey = "";
+          let progressLineColor = "";
+          if (negative && gspDelta.delta >= 0) {
+            return null;
+          }
+          if (!negative && gspDelta.delta <= 0) {
+            return null;
+          }
+          if (-200 < gspDelta.delta && gspDelta.delta < -60) {
+            bucketColor = "border-ocf-delta-100";
+            progressLineColor = "bg-ocf-delta-100";
+            dataKey = "-4";
+          } else if (-60 < gspDelta.delta && gspDelta.delta < -40) {
+            bucketColor = "border-ocf-delta-200";
+            progressLineColor = "bg-ocf-delta-200";
+            dataKey = "-3";
+          } else if (-40 < gspDelta.delta && gspDelta.delta < -20) {
+            bucketColor = "border-ocf-delta-300";
+            progressLineColor = "bg-ocf-delta-300";
+            dataKey = "-2";
+          } else if (-20 < gspDelta.delta && gspDelta.delta < -1) {
+            bucketColor = "border-ocf-delta-400";
+            progressLineColor = "bg-ocf-delta-400";
+            dataKey = "-1";
+          } else if (-1 <= gspDelta.delta && gspDelta.delta < 2) {
+            bucketColor = "border-white border-opacity-40";
+            progressLineColor = "bg-white bg-opacity-40";
+            dataKey = "0";
+          } else if (2 < gspDelta.delta && 20 > gspDelta.delta) {
+            bucketColor = "border-ocf-delta-600";
+            progressLineColor = "bg-ocf-delta-600";
+            dataKey = "1";
+          } else if (20 < gspDelta.delta && 40 > gspDelta.delta) {
+            bucketColor = "border-ocf-delta-700";
+            progressLineColor = "bg-ocf-delta-700";
+            dataKey = "2";
+          } else if (40 < gspDelta.delta && 60 > gspDelta.delta) {
+            bucketColor = "border-ocf-delta-800";
+            progressLineColor = "bg-ocf-delta-800";
+            dataKey = "3";
+          } else if ((60 < gspDelta.delta && 80 > gspDelta.delta) || gspDelta.delta > 80) {
+            bucketColor = "border-ocf-delta-900";
+            progressLineColor = "bg-ocf-delta-900";
+            dataKey = "4";
+          }
+
+          const isSelectedGSP = () => {
+            setClickedGspId(gspDelta.gspId);
+            setSelectedDeltaGsp(gspDelta.gspId);
```

Comment:
at this point they're doing the same thing, so I've removed the second one. I'd set up a global state variable to select the gsp in the table because that's how I was able to understand the logic.